src/frontend/Navigation/ScreenGroups/AppScreens.tsx

```diff
@@ -135,42 +137,48 @@ export type AppList = {
 const Tab = createBottomTabNavigator<HomeTabsList>();
 
 const HomeTabs = () => {
-  const locationState = useLocation({maxDistanceInterval: 0});
-  const [permissions] = useForegroundPermissions();
-  const locationProviderStatus = useLocationProviderStatus();
-
-  const precision = locationState.location?.coords.accuracy;
-
-  const locationStatus =
-    !!locationState.error || !permissions?.granted
-      ? 'error'
-      : getLocationStatus({
-          location: locationState.location,
-          providerStatus: locationProviderStatus,
-        });
+  const {handleTabPress} = useCurrentTab();
 
   return (
     <Tab.Navigator
+      screenListeners={{tabPress: handleTabPress}}
```

Comment:
Hm.. I don't get you, could you explain please what you mean, please?

Reply:
think this is no longer relevant, but the original idea was to do something like `screenListeners={{state: doSomething}}` so that you're deriving based on react navigation's internal state. but again, think it's not worth exploring right now given the other changes and conclusions elsewhere :)